types/types.go

```diff
@@ -23,67 +24,177 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type State uint8
+
+const (
+	Unprocessed State = iota
+	Active
+	Silenced
+	Inhibited
+)
+
+// AlertStatus stores the state and values associated with an Alert.
+type AlertStatus struct {
+	status State
+	values []string
```

Comment:
`AlertStatus` implements the marshaler interface so we can convert the enum into a string representation, so that handles marshaling the unexported fields.

I flatten the `AlertStatus` inside `APIAlert` in `dispatch.go:139`, but I'm not sure if that's the "right" way to do it.